### PR TITLE
fix: add missing timeseries expiration handling

### LINKED ISSUE #11229: expireAfterSeconds schema option not valid and not effective on timeseries collection
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
I use moongose for create timeseries collection
my schema is like this : 
```
const schema = new mongoose.Schema({ timestamp: Date, metadata: metricMetadataSchema },
  {
    _id: false,
    strict: false,
    autoIndex: false,
    autoCreate: false,
    timeseries: {timeField: "timestamp", metaField: "metadata", granularity: 'minutes'},
    // @ts-ignore
    expireAfterSeconds: 60
  }
)
```
At this point the property expireAfterSeconds is not valid for typescript because expireAfterSeconds not include in type SchemaOptions
The collection is well  done created, but no info about expireAfterSeconds anywhere  and after 60 seconds data not removed.

**If the current behavior is a bug, please provide the steps to reproduce.**

```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "es2017",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true
  }
}
```
**What is the expected behavior?**

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
node: v14.17.0
mongodb: 4.3.0
mongoose: 6.1.6

### PR DESCRIPTION
**Pull request by Uzlopak**
Resolves #11229 

Didnt know if I should just add expiresAfterSeconds which does only pass through the value to createCollection or like we do it in indexes with expires where we also handle date strings. So I implemented both. Removing is easier than adding.

Added unit test

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -1301,6 +1301,12 @@ declare module 'mongoose' {
     /** The timeseries option to use when creating the model's collection. */
     timeseries?: mongodb.TimeSeriesCollectionOptions;
 
+    /** The number of seconds after which a document in a timeseries collection expires. */
+    expireAfterSeconds?: number;
+
+    /** The time after which a document in a timeseries collection expires. */
+    expires?: number | string;
```
Comment (simonbrunel) [inline types/index.d.ts:1308]: Could make this even safer by using the [StringValue  type provided by the ms lib](https://github.com/vercel/ms/blob/master/src/index.ts#L44):

```ts
import type { StringValue as MsStringValue } from 'ms';

// ...

/** The time after which a document in a timeseries collection expires. */
expires?: number | MsStringValue;
```

Comment (Uzlopak) [reply types/index.d.ts:1308]: That ist good to know. @vkarpov15 should give the Feedback how he wants to set the expiration and then I would also fix that with the last commit.

**Comment by Uzlopak** [comment]
Maybe instead of naming it expiresAfterSeconds or expires calling it timeseriesExpires is better?

File: types/index.d.ts (hunk 1 above)
Comment (Uzlopak) [reply types/index.d.ts:1308]: I checked it. ms does not provide typings...

Comment (simonbrunel) [reply types/index.d.ts:1308]: Indeed, types will be available in [version 3](https://github.com/vercel/ms#typescript-support).

File: lib/model.js [hunk 2]
```diff
@@ -1375,6 +1375,12 @@ Model.createCollection = function createCollection(options, callback) {
   this.schema.options.timeseries;
   if (timeseries != null) {
     options = Object.assign({ timeseries }, options);
+    if (this.schema.options.expireAfterSeconds) {
```
Comment (vkarpov15) [inline lib/model.js:1378]: Let's add a check for `if (options.expireAfterSeconds != null)`. We don't want the schema option to override user-provided options if someone calls `createCollection({ expireAfterSeconds })`.

So I think this should look like:

```javascript
if (options.expireAfterSeconds != null) {
  // ...
} else if (options.expires != null) {
  // ..
} else if (this.schema.options.expireAfterSeconds != null) {
  // ...
} else if (this.schema.options.expires != null) {
  // ...
}
```

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

File: lib/model.js (hunk 2 above)
Comment (Uzlopak) [reply lib/model.js:1378]: Note to myself: I guess typings of parameter options of createCollection has also be adapted.

**Comment by Uzlopak** [comment]
@vkarpov15 

Adapted changes. Added unit and typings tests. Changed typings.

File: lib/model.js (hunk 2 above)
Comment (Uzlopak) [reply lib/model.js]: done

**Review by vkarpov15 — APPROVED** [review]
Thanks :+1:

**Pull request merged**